sympy/matrices/expressions/matmul.py

```diff
@@ -10,9 +10,10 @@
 from sympy.matrices.expressions.matexpr import (MatrixExpr, ShapeError,
         Identity, ZeroMatrix)
 from sympy.matrices.matrices import MatrixBase
+from sympy.core.operations import AssocOp
 
 
-class MatMul(MatrixExpr):
+class MatMul(MatrixExpr, AssocOp):
```

Comment:
Consider for instance `MatMul(MatrixSymbol("A", n, m), MatrixSymbol("B", m, k))`. 

Even ignoring that, one of the reasons AssocOp wants an identity is so that things like `Mul()` with no arguments could return the identity. But there's no way for `MatMul()` to return an identity (it needs a shape).